src/main/java/net/coderbot/iris/mixin/MixinEnderDragonEntityRenderer.java

```diff
@@ -0,0 +1,23 @@
+package net.coderbot.iris.mixin;
+
+import net.coderbot.iris.uniforms.CapturedRenderingState;
+import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.render.entity.EnderDragonEntityRenderer;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.entity.boss.dragon.EnderDragonEntity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(EnderDragonEntityRenderer.class)
+public class MixinEnderDragonEntityRenderer {
+	@Inject(method = "render", at = @At("HEAD"))
+	private void setTicksSinceDeath(EnderDragonEntity enderDragonEntity, float f, float g, MatrixStack matrixStack, VertexConsumerProvider vertexConsumerProvider, int i, CallbackInfo ci) {
+		if (enderDragonEntity.ticksSinceDeath != 0) {
+			CapturedRenderingState.INSTANCE.setTicksSinceDragonDeath(enderDragonEntity.ticksSinceDeath);
```

Comment:
This is a very undesirable way of doing this since the uniform won't have a valid value until the middle of the frame. Also, it doesn't handle multiple ender dragons either, it could be sane to define this as the maximum death progress of any ender dragon alive.

Reply:
This is the method vanilla seems to use when rendering, should we just rename the uniform to be in line with this?

Reply:
About the multiple dragon issue, this should never be an issue realistically.